src/Cnab.Bundle/Parameter.cs

```diff
@@ -6,19 +7,36 @@
 namespace Cnab.Bundle
 {
     [JsonConverter(typeof(ParameterConverter))]
-    public abstract class Parameter
+    public interface IParameter
+    {
+        string DataType { get; set; }
+        bool Required { get; set; }
+        ParameterMetadata Metadata { get; set; }
+        Location Destination { get; set; }
+        bool IsValid(object value);
+    }
+
+    public abstract class Parameter<T> : IParameter
     {
         [JsonProperty("type")]
         public string DataType { get; set; }
 
         [JsonProperty("required")]
         public bool Required { get; set; }
 
+        [JsonProperty("defaultValue")]
+        public T DefaultValue { get; set; }
+
+        [JsonProperty("allowedValues")]
+        public List<T> AllowedValues { get; set; }
+
         [JsonProperty("metadata")]
         public ParameterMetadata Metadata { get; set; }
 
         [JsonProperty("destination")]
         public Location Destination { get; set; }
+
+        public abstract bool IsValid(object value);
```

Comment:
I gave this a lot thought - and I couldn't come up with a better solution for this.
(Essentially, you might want to parse "80" as integer for an `IntParameter`).

@itowlson - WDYT?

Reply:
For reference, the Go implementation also passes an empty interface for this.

Reply:
I'd suggest that a `Validate()` method that could return diagnostic information about _why_ the value is invalid would be more useful than one that returns a plain boolean.  What is the UI to do with a boolean?

Reply:
And would it be idiomatic C# to return a tuple with a boolean and a message?

Reply:
That would be okay in C# 7, but would feel awkward for C# 6 consumers.  The traditional way to do this would be to either:

* Define a wee struct `ValidationResult { bool Succeeded; string FailureMessage; }`; or
* Use an out parameter `bool Validate(object value, out string failureMessage)` (but this one is ugh too)

Personally I would define the wee struct, but the tuple is probably okay too - but not everyone is on C# 7.  (It looks like this library itself is mostly avoiding C# 6 and 7 syntax features though I'm not sure if that's intentional.)